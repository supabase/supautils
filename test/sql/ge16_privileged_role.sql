-- non-superuser privileged role
set role privileged_role;
\echo

-- non-superuser non-privileged role cannot manage bypassrls role attribute
set role rolecreator;

-- the error message changed in PG14
do $$
begin
  create role r bypassrls;
  exception when insufficient_privilege then null;
end;
$$ language plpgsql;

create role r;
alter role r nobypassrls;
alter role r bypassrls;
drop role r;

set role privileged_role;
\echo

-- member of privileged_role can do privileged role stuff
set role privileged_role_member;

grant testme to authenticator;

set role privileged_role;
\echo

-- privileged_role can do GRANT <role> to <reserved_role>
grant testme to authenticator;
\echo
