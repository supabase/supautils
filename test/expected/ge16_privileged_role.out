-- non-superuser privileged role
set role privileged_role;
\echo

-- non-superuser non-privileged role cannot manage bypassrls role attribute
set role rolecreator;
-- the error message changed in PG14
do $$
begin
  create role r bypassrls;
  exception when insufficient_privilege then null;
end;
$$ language plpgsql;
create role r;
alter role r nobypassrls;
ERROR:  permission denied to alter role
DETAIL:  Only roles with the BYPASSRLS attribute may change the BYPASSRLS attribute.
alter role r bypassrls;
ERROR:  permission denied to alter role
DETAIL:  Only roles with the BYPASSRLS attribute may change the BYPASSRLS attribute.
drop role r;
set role privileged_role;
\echo

-- member of privileged_role can do privileged role stuff
set role privileged_role_member;
grant testme to authenticator;
set role privileged_role;
\echo

-- privileged_role can do GRANT <role> to <reserved_role>
grant testme to authenticator;
NOTICE:  role "authenticator" has already been granted membership in role "testme" by role "privileged_role"
\echo

